backend/internal/repository/patients.go

```diff
@@ -1,37 +1,61 @@
 package repository
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/radiant-network/radiant-api/internal/types"
 	"gorm.io/gorm"
 )
 
+type Patient = types.Patient
+
 type PatientsRepository struct {
 	db *gorm.DB
 }
 
 type PatientsDAO interface {
-	GetPatientByOrganizationId(organizationCode string, organizationPatientId string) (*types.Patient, error)
+	GetPatientByOrganizationPatientId(organizationId int, organizationPatientId string) (*Patient, error)
+	GetPatientByOrgCodeAndOrgPatientId(organizationCode string, organizationPatientId string) (*Patient, error)
+	CreatePatient(newPatient *Patient) error
 }
 
 func NewPatientsRepository(db *gorm.DB) *PatientsRepository {
 	return &PatientsRepository{db: db}
 }
 
-func (r *PatientsRepository) GetPatientByOrganizationPatientId(organizationId int, organizationPatientId string) (*types.Patient, error) {
-	var patient types.Patient
+func (r *PatientsRepository) GetPatientByOrganizationPatientId(organizationId int, organizationPatientId string) (*Patient, error) {
+	var patient Patient
 	tx := r.db.
-		Table("patient").
+		Table(patient.TableName()).
```

Comment:
the table will be the one throw starrocks federation (radiant_jdbc.public.patient) not directly the table name so probably better to keep `patient` until we decide how to deal with the different table names